include/deal.II/lac/solver_gmres.h

```diff
@@ -2079,6 +2332,63 @@ void SolverFGMRES<VectorType>::solve(const MatrixType         &A,
                 SolverControl::NoConvergence(accumulated_iterations, res));
 }
 
+
+
+//----------------------------------------------------------------------//
+
+
+
+template <typename VectorType>
+DEAL_II_CXX20_REQUIRES(concepts::is_vector_space_vector<VectorType>)
+SolverFGMRES<VectorType>::SolverFGMRES(SolverControl            &cn,
+                                       VectorMemory<VectorType> &mem,
+                                       const AdditionalData     &data)
+  : SolverMPGMRES<VectorType>(
+      cn,
+      mem,
+      typename SolverMPGMRES<VectorType>::AdditionalData{
+        data.max_basis_size,
+        true,
+        data.orthogonalization_strategy})
```

Comment:
If you derive from the `AdditionalData` class of `MPSolverGMRES`, you can just pass `data` to the base class's constructor here.

Reply:
I am not sure I want to because the boolean `use_truncated_mpgmres_strategy` does not make much sense *semantically* for `SolverFGMRES`. And there is no good way of hiding the parameter with public inheritance.

Reply:
You can't "hide" the parameter, but aren't all possible values equivalent for the case of only one preconditioner?